src/main/java/application/IncompleteSessionController.java

```diff
@@ -0,0 +1,132 @@
+package application;
+
+import auth.DjangoAuth;
+import beans.*;
+import exceptions.FormNotFoundException;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import objects.SerializableFormSession;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.commcare.modern.database.TableBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.web.bind.annotation.*;
+import repo.FormSessionRepo;
+import session.FormSession;
+import util.Constants;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+
+/**
+ * Controller class (API endpoint) containing all incomplete session management commands
+ *
+ */
+@Api(value = "Incomplete Session Controller", description = "Operations for navigating managing incomplete sessions")
+@RestController
+@EnableAutoConfiguration
+public class IncompleteSessionController extends AbstractBaseController{
+
+    @Value("${commcarehq.host}")
+    private String host;
+
+    private final Log log = LogFactory.getLog(IncompleteSessionController.class);
+
+    @Autowired
+    @Qualifier(value = "migrated")
+    protected FormSessionRepo migratedFormSessionRepo;
+
+    @ApiOperation(value = "Open an incomplete form session")
+    @RequestMapping(value = Constants.URL_INCOMPLETE_SESSION , method = RequestMethod.POST)
+    public NewFormResponse openIncompleteForm(@RequestBody IncompleteSessionRequestBean incompleteSessionRequestBean,
+                                              @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken) throws Exception {
+        log.info("Incomplete session request with bean: " + incompleteSessionRequestBean + " sessionId :" + authToken);
+        SerializableFormSession session;
+        restoreFactory.configure(incompleteSessionRequestBean, new DjangoAuth(authToken));
+        try {
+            session = formSessionRepo.findOneWrapped(incompleteSessionRequestBean.getSessionId());
+        } catch(FormNotFoundException e) {
+            session = migratedFormSessionRepo.findOneWrapped(incompleteSessionRequestBean.getSessionId());
+            // Move over to formplayer db
+            formSessionRepo.save(session);
```

Comment:
👍  good call. Would've worked this way too since we'd always find the formplayer version first, but agreed this is much cleaner.